specs/rollup-node.md

```diff
@@ -61,6 +66,46 @@ byte hash corresponding to the [L2 output root](./proposals.md#l2-output-commitm
 
 [SSZ]: https://github.com/ethereum/consensus-specs/blob/dev/ssz/simple-serialize.md
 
+### Structures
+
+These define the types used by rollup node API methods.
+The types defined here are extended from the [engine API specs][engine-structures].
+
+#### BlockID
+
+- `hash`: `DATA`, 32 Bytes
+- `number`: `QUANTITY`, 64 Bits
+
+#### L1BlockRef
+
+- `hash`: `DATA`, 32 Bytes
+- `number`: `QUANTITY`, 64 Bits
+- `parentHash`: `DATA`, 32 Bytes
+- `timestamp`: `QUANTITY`, 64 Bits
+
+#### L2BlockRef
+
+- `hash`: `DATA`, 32 Bytes
+- `number`: `QUANTITY`, 64 Bits
+- `parentHash`: `DATA`, 32 Bytes
+- `timestamp`: `QUANTITY`, 64 Bits
+- `l1origin`: `BlockID`
+- `sequenceNumber`: `QUANTITY`, 64 Bits - distance to first block of epoch
+
+#### SyncStatus
+
+Represents a snapshot of the rollup driver.
+
+- `current_l1`: `Object` - instance of [`L1BlockRef`](#l1blockref).
+- `current_l1_finalized`: `Object` - instance of [`L1BlockRef`](#l1blockref).
```

Comment:
There's also `finalized_l1` which is different somehow and even the comment in the code doesn't really help me understand what's different about it.

Reply:
Somehow missed that. The `current_finalized_l1` is the derivation's own state of finalized L1. It's updated as derivation ingests blocks. Whereas, `finalized_l1` is the latest L1 state observed, but not processed, by derivation.